static/js/filter.js

```diff
@@ -31,16 +37,29 @@
         
         if (input.target) {
             query = input.target.value.toLowerCase();
-        }   
+        }
+
+        filterInput.removeEventListener("focus", filterFocusWatcher, false);
 
         // Reset filter if the input is empty, however, do not steal focus to input
         if (query === "") resetFilter();
 
         // Add class to keep the reset button visible while a query has been entered
         filterInput.classList.add("is-filtered");
 
-		// Hide all cases
-		aliases.forEach(alias => {
+        const isCategoryFilterActive = (filterContainer.classList.contains("is-filtered-by-category"));
+
+        currentFilteredByCategoryAliases = aliases;
+
+        if (isCategoryFilterActive) {
+            // aliases = 
+            currentFilteredByCategoryAliases = document.querySelectorAll(".c-alias:not(.is-hidden)");
+        } else {
+            filterContainer.classList.add("is-filtered-by-search");
+        }
```

Comment:
This logic block appears throughout this PR and is the crux of the overall feature. Basically when either search or category filtering is active, we have to trunicate the list of available aliases for the OTHER filter to query from. Moving to state management would drop this logic, giving each filtering type the same centralized "target" to query from.